src/R2.jl

```diff
@@ -86,6 +86,12 @@ end
   return solve!(solver, nlp; kwargs...)
 end
 
+function SolverCore.reset!(solver::R2Solver{T}) where T
+  solver.d .= zero(T)
```

Comment:
i remember now why I suggest to parametrize with `T` only. You use it at line 90.
`V` is not used so it's not relevant to keep it.

Reply:
Ok, but the type `R2Solver{T]` doesn't exist, it is `R2Solver{T, V}`

Reply:
Yes, ok. But we lose in readability. Is there an interest in doing that?

Reply:
We don't ask the compiler to generate a new method if `V` is different. If it's not relevant to compile a new method, you shoud not over-parameterize.